src/dbnode/client/session.go

```diff
@@ -966,8 +966,8 @@ func (s *session) writeAttemptWithRLock(
 	// use in the various queues. Tracking per writeAttempt isn't sufficient as
 	// we may enqueue multiple writeStates concurrently depending on retries
 	// and consistency level checks.
-	nsID := s.pools.id.Clone(namespace)
-	tsID := s.pools.id.Clone(id)
+	nsID := s.cloneFinalizable(namespace)
```

Comment:
nit: also add a test cases to ensure we're doing the right thing. https://github.com/m3db/m3/blob/master/src/x/test/util.go#L30 won't be for naught